transactor/src/main/scala/io/mediachain/transactor/StateMachine.scala

```diff
@@ -25,15 +28,23 @@ object StateMachine {
   case class JournalLookup(
     ref: Reference
   ) extends Query[Option[Reference]]
+
+  case class JournalCurrentBlock() extends Query[JournalBlock]
   
-  case class JournalCommitEvent(entry: JournalEntry) extends Serializable
+  sealed abstract class JournalEvent extends Serializable
+  case class JournalCommitEvent(entry: JournalEntry) extends JournalEvent
+  case class JournalBlockEvent(ref: Reference) extends JournalEvent
 
   class JournalStateMachine(
-    val datastore: Datastore
+    val datastore: Datastore,
+    val blocksize: Int = JournalBlockSize // configurable to facilitate testing
   ) extends CopycatStateMachine with Snapshottable with SessionListener {
     private var seqno: BigInt = 0
     private var index: MMap[Reference, ChainReference] = new MHashMap // canonical -> chain map
-    private val clients: MSet[ServerSession] = new MHashSet // this wanted to be called sessions
+    private var block: ListBuffer[JournalEntry] = new ListBuffer      // current block entries
```

Comment:
I'll change to mutable.Buffer, seems like a better class for the purpose (I didn't know it existed :)